dapp_runner/runner/runner.py

```diff
@@ -83,6 +86,49 @@ async def _start_local_proxy(
             {name: {LOCAL_PROXY_DATA_KEY: f"http://localhost:{port}"}}
         )
 
+    async def _start_service(
+        self, service_name: str, service_descriptor: ServiceDescriptor
+    ):
+        # if this service depends on another, wait until the dependency is up
+        if service_descriptor.depends_on:
+            while (
+                service_descriptor.depends_on not in self.clusters
+                or not self._is_cluster_state(
+                    service_descriptor.depends_on, ServiceState.running
+                )
+            ):
+                await asyncio.sleep(DEPENDENCY_WAIT_INTERVAL)
+
+        cluster_class, run_params = await get_service(
+            service_name, service_descriptor, self._payloads, self._networks
+        )
+        cluster = await self.start_cluster(service_name, cluster_class, run_params)
+
+        if service_descriptor.http_proxy:
+            # wait until the service is running before starting the proxy
+            while not self._is_cluster_state(service_name, ServiceState.running):
```

Comment:
I don't like the fact that:
* `_start_service` for services without `http_proxy` finishes quite fast (without any assumption about service state)
* `_start_service` for services with `http_proxy` finishes only after we have a `Running` service.

I've a feeling that this might lead to a lot of confusion/complex bugs.

Few solutions seem possible:
A) Extract the contents of this `if` to a separate task (that will wait for the `running` state and start the local proxy later
B) Just ignore the state and start proxy, accepting that it will not be operative immediately

A seems better, but I'm not sure about the downsides.
I consider this pretty important, so if this change is as easy as it seems to me then please change this :)

Reply:
hmm, good points, I'll think about it for a short while... I think I'd go with A ...

Reply:
:+1: